autogen/agentchat/contrib/capabilities/transforms.py

```diff
@@ -200,6 +202,114 @@ def _validate_max_tokens(self, max_tokens: Optional[int] = None) -> Optional[int
         return max_tokens if max_tokens is not None else sys.maxsize
 
 
+class TextMessageCompressor:
+    """A transform for compressing text messages in a conversation history.
+
+    It uses a specified text compression method to reduce the token count of messages, which can lead to more efficient
+    processing and response generation by downstream models.
+    """
+
+    def __init__(
+        self, text_compressor: Optional[TextCompressor] = None, compress_all_messages_on_first_apply: bool = False
+    ):
+        """
+        Args:
+            text_compressor (TextCompressor or None): An instance of a class that implements the TextCompressor protocol. If None, it defaults to LLMLingua.
+            compress_all_messages_on_first_apply (bool): Whether to apply the text compression to all messages initially. Defaults to False.
+        """
+
+        if text_compressor is None:
+            text_compressor = LLMLingua()
+
+        self._text_compressor = text_compressor
+        self._init_all_messages = compress_all_messages_on_first_apply
+        self._transform_applied_once = False
```

Comment:
```
self._compress_all_on_first = compress_all_on_first
self._to_compress_all = compress_all_on_first # this will be set to False after the first application
```
These names I think make the code's structure and objectives clearer.

Also:
* line 244 will be simplified to `if self._to_compress_all:`
* line 249 can be moved within the if statement (changed to `self._to_compress_all = False`). This way, it's associated with the specific condition of the if statement and not the entire function.